python/python/bystro/prs/tests/test_prs_dosage.py

```diff
@@ -0,0 +1,277 @@
+
+    # Simulate Polygenic Risk Score Calculation
+    prs_scores = loaded_arrays.multiply(weights, axis="index").sum()
+
+    return prs_scores.to_dict()
+
+
+def test_prs_calculations_with_pandas(create_feather_file):
+    """
+    Test PRS calculations with Pandas
+    This will be less memory efficient than operating on the Feather file directly
+    """
+    print("running test_prs_calculations_with_pandas")
+    file_path = create_feather_file
+
+    prs_scores = calculate_prs_in_pandas(file_path, weights)
+    assert len(prs_scores) == len(table.column_names) - 1  # excluding 'locus' column
+    print(prs_scores)
+
+
+def test_prs_calculations_arrow_with_missing_values(create_feather_file):
+    """
+    Test PRS calculations when the genotype weights have loci not in the genotype dosage matrix
+    """
+    file_path = create_feather_file
+
+    # Read the Feather file into a PyArrow Table
+    with pa.memory_map(file_path, "r") as source:
+        table = pa.ipc.open_file(source).read_all()
+
+    # Compute the weighted sum for each sample column
+    prs_scores = {}
+    weights_subsetted_to_valid = None
+    for column_name in table.column_names:
+        valid_loci = []
+        if column_name == "locus":
+            valid_loci = table.column(column_name).to_pylist()
+            weights_subsetted_to_valid = weights_with_missing.loc[valid_loci]
+
+        if column_name != "locus":
+            sample_column = table.column(column_name)
+            weighted_values = pc.multiply(sample_column, pa.array(weights_subsetted_to_valid))
+            prs_scores[column_name] = pc.sum(weighted_values).as_py()
+
+    assert len(prs_scores) == len(table.column_names) - 1  # excluding 'locus' column
+    print(prs_scores)
+
+
+def calculate_prs_in_batches(file_path, weights):
+    """
+    Calculate PRS scores efficiently, in batches
+    """
+
+    my_dataset = ds.dataset([file_path], format="arrow")
+    mask = pc.field("locus").isin(weights.index)
+    my_dataset = my_dataset.filter(mask)
+
+    # Read the Feather file into a Pandas DataFrame in batches
+    # and calculate PRS in batches
+    prs_scores: dict[str, float] = {}
+    for batch in my_dataset.to_batches():
+        loaded_arrays = batch.to_pandas()
+        loaded_arrays = loaded_arrays.set_index("locus")
+        weights_subset = weights.loc[loaded_arrays.index]
+        prs_scores_batch = loaded_arrays.multiply(weights_subset, axis="index").sum()
+        # add the batch scores to the total scores for each sample
+        for sample in prs_scores_batch.index:
+            if sample in prs_scores:
+                prs_scores[sample] += prs_scores_batch[sample]
+            else:
+                prs_scores[sample] = prs_scores_batch[sample]
+
+    return prs_scores
+
+
+def test_prs_one_batch(create_feather_file):
+    """
+    Test PRS calculations with Pandas
+    This will be less memory efficient than operating on the Feather file directly
```

Comment:
If you want to include some kind of performance testing here, I think it would be useful to have a performance metric as the output vs printing the scores if the scores are going to be the same between methods that you're testing.

Reply:
>  if the scores are going to be the same between methods that you're testing.

We need to test that is the case, which is why we do `assert prs_scores == prs_scores_pandas`, because it is not guaranteed that we handle batching correctly. 

I can drop all print statements. There is a better way of handling benchmarking, and I can add that in.